exercises/grep/canonical-data.json

```diff
@@ -332,6 +363,53 @@
             "files": ["iliad.txt", "midsummer-night.txt", "paradise-lost.txt"]
           },
           "expected": []
+        },
+        {
+          "description": "Multiple files, several matches, file flag takes precedence over line number flag",
+          "property": "grep",
+          "input": {
+            "pattern": "who",
+            "flags": ["-n", "-l"],
+            "files": ["iliad.txt", "midsummer-night.txt", "paradise-lost.txt"]
+          },
+          "expected": [
+            "iliad.txt",
+            "paradise-lost.txt"
+          ]
+        },
+        {
+          "description": "Multiple files, several matches, inverted and match entire lines flags",
+          "property": "grep",
+          "input": {
+            "pattern": "Illustrious into Ades premature,",
+            "flags": ["-x", "-v"],
+            "files": ["iliad.txt", "midsummer-night.txt", "paradise-lost.txt"]
+          },
+          "expected": [
+            "Achilles sing, O Goddess! Peleus' son;",
```

Comment:
notice that in the above `multiple files` tests, the lines are each prefixed with the file they appear in.

I believe the same should happen here, as demonstrated in an experiment I share below:

```
$ grep -x -v hello <(echo "hello\nworld") <(echo "a\nhello") <(echo "hello\nhello")
/proc/self/fd/11:world
/proc/self/fd/12:a
```